packages/matter.js/test/behavior/cluster/ClusterBehaviorTest.ts

```diff
@@ -29,12 +29,29 @@ import { My, MyBehavior, MyCluster } from "./cluster-behavior-test-util.js";
 class MyEventedBehavior extends MyBehavior {
     declare events: MyEventedBehavior.Events;
 }
+
 namespace MyEventedBehavior {
     export class Events extends MyBehavior.Events {
         somethingHappened = new Observable<[]>();
     }
 }
 
+class BehaviorWithCustomMethods extends MyBehavior.with("Awesome") {
+    foo() {
+        return true;
+    }
+
+    bar() {
+        return 4;
+    }
+}
+
+namespace BehaviorWithCustomMethods {
+    export declare const ExtensionInterface: {
```

Comment:
Can this also being extended?

So BehaviorWithCustomMethods adds bar() ..

Now I extend all this and have BehaviorWithEvenMoreCustomMethods which adds fun() ... ?

Reply:
Yes you'd have to use an intersection type.  Hopefully wouldn't happen very much:

```typescript
namespace BehaviorWithEvenMoreCustomMethods {
    export declare const ExtensionInterface: BehaviorWithCustomMethods & {
        fun(): boolean;
    }
}
```